test/bundle_name_test.js

```diff
@@ -0,0 +1,184 @@
+
+		bundle = {
+			bundles: ["main/bar", "bar/foo"],
+			buildType: "txt"
+		};
+		bundleName = nameBundle.getName(bundle);
+		assert.equal(bundleName, dirName + 'bar-foo.txt!');
+	});
+
+	it("works if buildType is a extension", function(){
+		var bundle = {
+			bundles: ["main"],
+			buildType: "txt"
+		};
+		nameBundle(bundle);
+		assert.equal(bundle.buildType, 'txt');
+		assert.equal(bundle.name, dirName + 'main.txt!');
+	});
+
+	it("with very long name", function(){
+		var count = 0;
+
+		var bundle = {
+			bundles: ["deep/folder/structure/with/very/very-very-long-name"]
+		};
+		var bundleName = nameBundle.getName(bundle);
+		assert.equal(bundleName, dirName + 'deep/folder/structure/with/very/very-very-long-name');
+
+		// again...
+		bundleName = nameBundle.getName(bundle);
+		assert.equal(bundleName, dirName + 'deep/folder/structure/with/very/very-very-long-nam-1-' + (count++));
+
+		bundle = {
+			bundles: ["main/my-very-long-name", "bar/eman-gnol-yrev-ym"]
+		};
+		bundleName = nameBundle.getName(bundle);
+		assert.equal(bundleName, dirName + 'my-very-long-nam-34e05993');
+
+
+		bundle = {
+			bundles: ["deep/folder/structure/with/very/very/very-long-name"],
+			buildType: "txt"
+		};
+		nameBundle(bundle);
+		assert.equal(bundle.buildType, 'txt');
+		assert.equal(bundle.name, dirName + 'deep/folder/structure/with/very/very/very-long-name.txt!');
+
+		// again...
+		nameBundle(bundle);
+		assert.equal(bundle.name, dirName + 'deep/folder/structure/with/very/very/very-long-nam-1-'+ (count++) +'.txt!');
+
+	});
+
+	it("works with a plugin markup", function() {
+		var bundle = {
+			bundles: ["index.stache!done-autorender"]
+		};
+		var bundleName = nameBundle.getName(bundle);
+		assert.equal(bundleName, dirName + 'index');
+
+		bundle = {
+			bundles: ["main!my-plugin"]
+		};
+		bundleName = nameBundle.getName(bundle);
+		assert.equal(bundleName, dirName + 'main');
+	});
+
+});
+
+describe("bundle name with npm package", function() {
+	var oldWinstonWarning,
+		warns;
+	beforeEach(function () {
+		oldWinstonWarning = winston.warn;
+		warns = [];
+
+		winston.warn = function(msg){
+			warns.push(msg);
+		};
+	});
+
+	afterEach(function(){
+		winston.warn = oldWinstonWarning;
+	});
+
+	it("works for a package module", function(){
+		var bundle = {
+			bundles: ["foo@1.0.0#bar"]
+		};
+		var bundleName = nameBundle.getName(bundle);
+		assert.equal(bundleName, dirName + 'foo/bar');
+	});
+
+	it("works with multiple bundles", function(){
+		var bundle = {
+			bundles: ["foo@1.0.0#component/my-component", "foo@1.0.0#foo"]
+		};
+		var bundleName = nameBundle.getName(bundle);
+		assert.equal(bundleName, dirName + 'foo/my-component-foo');
+	});
+
+	it("remove trailing parts", function () {
+		var bundle = {
+			bundles: ["foo@1.0.0#main/bar/", "foo@1.0.0#foo.js"]
+		};
+		var bundleName = nameBundle.getName(bundle);
+		assert.equal(bundleName, dirName + 'foo/bar-foo');
+	});
+
+	it("works with a plugin markup", function(){
```

Comment:
This is the one I want to make sure is still working like it previously was.  I think that it is.